packages/p2p/src/components/orders/order-details/order-details-footer.jsx

```diff
@@ -1,188 +1,335 @@
+                                    target='_blank'
+                                    href={'mailto:p2p-support@deriv.com'}
+                                />,
+                            ]}
+                        />
+                        {localize(' team.')}
+                    </div>
+                </Modal.Body>
+                <Modal.Footer>
+                    {error_message && <FormError message={error_message} />}
+                    <Button.Group>
+                        <Button secondary type='button' onClick={hideComplainOrderModal} large>
+                            {localize('Cancel')}
+                        </Button>
+                        <Button is_disabled={!dispute_reason} primary large onClick={disputeOrderRequest}>
+                            {localize('Submit')}
+                        </Button>
+                    </Button.Group>
+                </Modal.Footer>
+            </Modal>
+        );
     };
 
-    const showReceivedFundsPopup = () => {
-        const sendOrderConfirmRequest = async setFormStatus => {
-            setFormStatus({ error_message: '' });
+    const ConfirmOrderModal = () => {
+        const [error_message, setErrorMessage] = React.useState('');
+        const [is_checkbox_checked, setIsCheckboxChecked] = React.useState(false);
 
-            const update_response = await requestWS({
-                p2p_order_confirm: 1,
-                id,
+        const confirmOrderRequest = () => {
+            return new Promise(resolve => {
+                requestWS({
+                    p2p_order_confirm: 1,
+                    id,
+                }).then(response => {
+                    if (isMounted()) {
+                        if (response.error) {
+                            setErrorMessage(response.error.message);
+                        }
+                    }
+                    resolve();
+                });
             });
-
-            if (!update_response.error) {
-                cancelPopup();
-            } else {
-                setFormStatus({ error_message: update_response.error.message });
-            }
         };
 
-        const options = {
-            title: localize('Have you received payment?'),
-            className: 'order-details__popup-no-border',
-            message: localize(
-                'Please confirm only after checking your bank or e-wallet account to make sure you have received payment.'
-            ),
-            need_confirmation: true,
-            order_information,
-            onClickConfirm: sendOrderConfirmRequest,
-            has_cancel: true,
-            cancel_text: localize('Cancel'),
-        };
+        return (
+            <Modal
+                className='confirm-modal'
+                is_open={should_show_confirm_modal}
+                toggleModal={hideConfirmOrderModal}
+                title={
+                    (is_buy_order && !is_my_ad) || (is_sell_order && is_my_ad)
```

Comment:
We're reusing this same condition throughout the component, could we create variables that better describe them?